src/script/page/state.ts

```diff
@@ -52,7 +80,152 @@ type AppMainState = {
   };
 };
 
+const legalHoldModalDefaultState = {
```

Comment:
This file already starts to be huge. We do not really need to have everything in a single state, we could have multiple separated and canonical states. 
The modal component could come with it's own state management and it could live next to the component. 
WDYT?